File: data_types_project/cpp/assoc_arrays.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;

int main() {
    map<string, int> scores;

    scores["Alice"] = 95;
    scores["Bob"] = 82;
    scores["Charlie"] = 74;

    for (auto pair : scores) {
        cout << pair.first << ": " << pair.second << endl;
    }

    return 0;
}

File: data_types_project/rust/assoc_arrays.rs
use std::collections::HashMap;

fn main() {
    let mut scores = HashMap::new();

    scores.insert("Alice", 95);
    scores.insert("Bob", 82);
    scores.insert("Charlie", 74);

    for (name, score) in &scores {
        println!("{}: {}", name, score);
    }
}